hazelcast/src/main/java/com/hazelcast/spi/impl/AbstractCompletableFuture.java

```diff
@@ -110,25 +163,71 @@ public V get() throws InterruptedException, ExecutionException {
         }
     }
 
+    @Override
+    public V get(long timeout, TimeUnit unit) throws InterruptedException, ExecutionException, TimeoutException {
+        final long deadlineTimeMillis = System.currentTimeMillis() + unit.toMillis(timeout);
+        long millisToWait;
+        for (; ; ) {
+            Object currentState = this.state;
+
+            if (isCancelledState(currentState)) {
+                throw new CancellationException();
+            }
+            if (isDoneState(currentState)) {
+                return getResult();
+            }
+            if (Thread.interrupted()) {
+                throw new InterruptedException();
+            }
+
+            millisToWait = deadlineTimeMillis - System.currentTimeMillis();
+            if (millisToWait <= 0) {
+                throw new TimeoutException();
+            }
+
+            synchronized (this) {
+                if (!isDoneState(this.state)) {
```

Comment:
I just hate admitting in a method name that we don't know the semantics of a "done" future :) Keeping the current name and adding a Javadoc which reminds of the meaning of "done" would sit nicer with me.